src/components/lib/address-link.tsx

```diff
@@ -0,0 +1,67 @@
+
+import Link from "next/link";
+import { SquareArrowOutUpRight } from "lucide-react";
+import { cn } from "@/lib/utils";
+import {
+  Tooltip,
+  TooltipContent,
+  TooltipProvider,
+  TooltipTrigger,
+} from "@/components/ui/tooltip";
+import useGlobalContext from "@/components/lib/context/hook";
+
+interface AddressLinkProps {
+  address: string | null;
+  href: string;
+  isExternal?: boolean;
+}
+
+const AddressLink = ({
+  address,
+  href,
+  isExternal = false
+}: AddressLinkProps) => {
+  const {
+    state: { hoveredAddress },
+    setHoveredAddress,
+  } = useGlobalContext();
+
+
+  return (
+    <TooltipProvider>
+      <Tooltip
+        delayDuration={100}
+        onOpenChange={(open) => setHoveredAddress((open && address) ? address : "")}
+      >
+        <TooltipTrigger>
+            <Link
+            href={href}
+            className={"flex items-center"}
```

Comment:
Here use `cn("flex items-center", className)` as `className` to merge the default `Link` classname along with the `className` prop passed by the parent component.